plugins/woocommerce/client/admin/client/settings-email/settings-email-preview-iframe.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState } from 'react';
+import { debounce } from 'lodash';
+
+type EmailPreviewIframeProps = {
+	src: string;
+	setIsLoading: ( isLoading: boolean ) => void;
+	settingsIds: string[];
+};
+
+export const EmailPreviewIframe: React.FC< EmailPreviewIframeProps > = ( {
+	src,
+	setIsLoading,
+	settingsIds,
+} ) => {
+	const [ counter, setCounter ] = useState( 0 );
+
+	useEffect( () => {
+		const handleFieldChange = async ( event: Event ) => {
+			const target = event.target as HTMLInputElement;
+			const key = target.id;
+			const value = target.value;
+
+			setIsLoading( true );
+
+			try {
+				await apiFetch( {
+					path: 'wc-admin-email/settings/email/save-transient',
+					method: 'POST',
+					data: { key, value },
+				} );
+			} finally {
+				setIsLoading( false );
```

Comment:
This makes the loading indicator disappear too early, before the frame starts loading. As a result, several seconds might pass between the loading indication and the actual frame refreshing, which may leave an impression that changes aren't applied. 

I think this line can be removed, so it will be triggered by the iframe's `onLoad`. But if this is needed for error handling, replacing `finally` with `catch` or making this line conditional (only executed if an error occurs) could be a solution.

Reply:
Great observation. This is indeed unnecessary at this place because iframe's onLoad will handle it. Removed in 4e1d407.